### DISCUSSION TITLE
How does kitty implement transparency with SGR color sequences?

### DISCUSSION THREAD

**Discussion by fxzzi** [kovidgoyal/kitty#6687] [Q&A] [answered]
Hi there everyone! I just wanted to ask a quick question about how kitty handles transparency with SGR color sequences. For example, if my background color in neovim is set to the same as my background colour in kitty, the background in neovim will also have my opacity applied to it. However in most other terminals like alacritty and foot, they render the exact same colour in vim as opaque.

How and where in code does kitty implement the opacity of SGR color sequences?

**Comment by kovidgoyal** [comment]
This is by design and is not going to change.
https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.background_opacity

**Reply by fxzzi** [reply]
Hey there,

I think you understood me wrong haha, I actually love how kitty handles it and I wanted to attempt to implement it myself in foot.

Is it [here](https://github.com/kovidgoyal/kitty/blob/f03027c33da2133cb2830898039105956d52127e/kitty/cell_vertex.glsl#L208) where kitty implements this behaviour? Sorry if I'm being a bit stupid, this is the first time I've looked at the kitty source code.

**Comment by kovidgoyal** [comment, marked as answer]
yes its implemented in a shader which I dont think foot uses, so not
sure what use the code is to you. Basically background_opacity is calculateds
as the setting_value * (1 if bgcolor == default_bg else 0)
